src/data/sym/card.lean

```diff
@@ -5,41 +5,212 @@ Authors: Yaël Dillies, Bhavik Mehta
 -/
 import algebra.big_operators.basic
 import data.finset.sym
+import tactic.slim_check
 
 /-!
 # Stars and bars
 
-In this file, we prove the case `n = 2` of stars and bars.
+In this file, we prove stars and bars.
 
 ## Informal statement
 
 If we have `n` objects to put in `k` boxes, we can do so in exactly `(n + k - 1).choose n` ways.
 
 ## Formal statement
 
-We can identify the `k` boxes with the elements of a fintype `α` of card `k`. Then placing `n`
+We can identify the `n` boxes with the elements of a fintype `α` of card `n`. Then placing `n`
 elements in those boxes corresponds to choosing how many of each element of `α` appear in a multiset
 of card `n`. `sym α n` being the subtype of `multiset α` of multisets of card `n`, writing stars
 and bars using types gives
 ```lean
--- TODO: this lemma is not yet proven
 lemma stars_and_bars {α : Type*} [fintype α] (n : ℕ) :
-  card (sym α n) = (card α + n - 1).choose (card α) := sorry
+  card (sym α n) = (card α + n - 1).choose n := sorry
 ```
 
-## TODO
-
-Prove the general case of stars and bars.
-
 ## Tags
 
 stars and bars
 -/
 
-open finset fintype
+open finset fintype function sum
+
+variables {α β : Type*}
+
+section to_move
+
+@[simp] lemma option.coe_get {o : option α} (h : o.is_some) : ((option.get h : α) : option α) = o :=
+option.some_get h
+
+end to_move
+
+section unused
+
+namespace multiset
+
+@[simp] lemma map_le_map_iff {f : α ↪ β} {s t : multiset α} : s.map f ≤ t.map f ↔ s ≤ t :=
```

Comment:
Similarly, these should either be moved or deleted.

Reply:
I've moved these to `data/multiset/basic` and `data/finset/basic`